fix torchdeploy issue on sharddim_alltoall op

wanchaol opened the pull request:
Summary: fix torchdeploy issues when registering the distributed op, similar to what functional collective did


Differential Revision: D56850434


cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang @d4l3k

XilunWu reviewed (approved):
thx for fixing the internal tests.

Commit:
fix torchdeploy issue on sharddim_alltoall op (#125344)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/125344

fix torchdeploy issues when registering the distributed op, similar to what functional collective did

Test Plan:
locally tested
```
buck2 test 'fbcode//mode/opt' fbcode//minimal_viable_ai/umia_v1/ig/gpu_tests:predictor_test -- --exact 'minimal_viable_ai/umia_v1/ig/gpu_tests:predictor_test - test_omni (minimal_viable_ai.umia_v1.ig.gpu_tests.predictor_test.PredictorTest)'
```

Reviewed By: yingufan, XilunWu

Differential Revision: D56850434
